command/agent/agent.go

```diff
@@ -222,6 +222,22 @@ func (a *Agent) setupClient() error {
 	conf.Node.Name = a.config.NodeName
 	conf.Node.Meta = a.config.Client.Meta
 	conf.Node.NodeClass = a.config.Client.NodeClass
+	httpAddr := a.config.BindAddr
+	if a.config.Addresses.HTTP != "" {
```

Comment:
Sorry had a typo. Just making it so that you don't resolve the address for addresses.HTTP when you are going to be using the AdvertiseAddrs.HTTP anyways.